packages/@aws-cdk/aws-codedeploy/lib/ecs/deployment-group.ts

```diff
@@ -55,8 +199,132 @@ export class EcsDeploymentGroup {
+      cfnService.deploymentController = { type: ecs.DeploymentControllerType.CODE_DEPLOY };
+
+      // Remove revision from ECS Service task definition to allow Blue/Green deploys to work
+      cfnService.taskDefinition = service.taskDefinition.family;
+      service.node.addDependency(service.taskDefinition);
+    }
+
+    // create a testTrafficRoute if missing
+    let testTrafficRoute = props.testTrafficRoute;
+    if (props.prodTrafficRoute && !testTrafficRoute) {
+      const prodCfnTargetGroup = props.prodTrafficRoute.targetGroup.node.defaultChild as CfnTargetGroup;
+      const testCfnTargetGroup = new CfnTargetGroup(this, 'TestTargetGroup', {
+        healthCheckEnabled: prodCfnTargetGroup.healthCheckEnabled,
+        healthCheckIntervalSeconds: prodCfnTargetGroup.healthCheckIntervalSeconds,
+        healthCheckPath: prodCfnTargetGroup.healthCheckPath,
+        healthCheckPort: prodCfnTargetGroup.healthCheckPort,
+        healthCheckProtocol: prodCfnTargetGroup.healthCheckProtocol,
+        healthCheckTimeoutSeconds: prodCfnTargetGroup.healthCheckTimeoutSeconds,
+        healthyThresholdCount: prodCfnTargetGroup.healthyThresholdCount,
+        ipAddressType: prodCfnTargetGroup.ipAddressType,
+        matcher: prodCfnTargetGroup.matcher,
+        name: prodCfnTargetGroup.name,
+        port: prodCfnTargetGroup.port,
+        protocol: prodCfnTargetGroup.protocol,
+        protocolVersion: prodCfnTargetGroup.protocolVersion,
+        targetGroupAttributes: prodCfnTargetGroup.targetGroupAttributes,
+        targets: prodCfnTargetGroup.targets,
+        targetType: prodCfnTargetGroup.targetType,
+        unhealthyThresholdCount: prodCfnTargetGroup.unhealthyThresholdCount,
+        vpcId: prodCfnTargetGroup.vpcId,
+      });
+      if (props.prodTrafficRoute.targetGroup instanceof ApplicationTargetGroup) {
+        testTrafficRoute = {
+          targetGroup: ApplicationTargetGroup.fromTargetGroupAttributes(this, 'TestTargetGroupRef', {
+            targetGroupArn: testCfnTargetGroup.ref,
+          }),
+          listener: undefined,
+        };
+      } else if (props.prodTrafficRoute.targetGroup instanceof NetworkTargetGroup) {
+        testTrafficRoute = {
+          targetGroup: NetworkTargetGroup.fromTargetGroupAttributes(this, 'TestTargetGroupRef', {
+            targetGroupArn: testCfnTargetGroup.ref,
+          }),
+          listener: undefined,
+        };
+      }
+    }
+
+    const resource = new CfnDeploymentGroup(this, 'Resource', {
+      applicationName: this.application.applicationName,
+      serviceRoleArn: this.role.roleArn,
+      deploymentGroupName: this.physicalName,
+      deploymentConfigName: this.deploymentConfig.deploymentConfigName,
+      deploymentStyle: (props.prodTrafficRoute && testTrafficRoute) ? {
+        deploymentType: 'BLUE_GREEN',
+        deploymentOption: 'WITH_TRAFFIC_CONTROL',
+      } : {
+        deploymentType: 'IN_PLACE',
+        deploymentOption: 'WITHOUT_TRAFFIC_CONTROL',
```

Comment:
This is not valid for ECS deployments.  CodeDeploy only supports BLUE_GREEN + WITH_TRAFFIC_CONTROL for ECS

Reply:
Removed.